typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,628 @@
+        if (!promiseGetServiceInfo.isOk()) {
+            return [null, promiseGetServiceInfo.error];
+        }
+
+        const serviceResponse: GetServiceInfoResponse = promiseGetServiceInfo.value;
+        if (serviceResponse.getIpAddr() === "") {
+            return [null, new Error(
+                "Kurtosis API reported an empty IP address for service " + serviceId +  " - this should never happen, and is a bug with Kurtosis!",
+                )
+            ];
+        }
+
+        const enclaveDataVolMountDirpathOnSvcContainer: string = serviceResponse.getEnclaveDataVolumeMountDirpath();
+        if (enclaveDataVolMountDirpathOnSvcContainer === "") {
+            return [null, new Error(
+                "Kurtosis API reported an empty enclave data volume directory path for service " + serviceId + " - this should never happen, and is a bug with Kurtosis!",
+                )
+            ];
+        }
+
+        const serviceContext: ServiceContext = new ServiceContext(
+            this.client,
+            serviceId,
+            serviceResponse.getIpAddr(),
+            this.enclaveDataVolMountpoint,
+            enclaveDataVolMountDirpathOnSvcContainer,
+        )
+
+        return [serviceContext, null];
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async removeService(serviceId: ServiceID, containerStopTimeoutSeconds: number): Promise<Error> {
+
+        winston.debug("Removing service '%v'...", serviceId);
+        // NOTE: This is kinda weird - when we remove a service we can never get it back so having a container
+        //  stop timeout doesn't make much sense. It will make more sense when we can stop/start containers
+        // Independent of adding/removing them from the network
+        const args: RemoveServiceArgs = newGetRemoveServiceArgs(serviceId, containerStopTimeoutSeconds);
+        
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING (REMOVE)
+        // if _, err := networkCtx.client.RemoveService(context.Background(), args); err !== nil {
+        //     return stacktrace.Propagate(err, "An error occurred removing service '%v' from the network", serviceId)
+        // }
+
+        const promiseAsyncRemoveService: Promise<ResultAsync<any, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.removeService(args, (_unusedError: Error, response: any) => {
+                resolve(okAsync(response));
+            })
+        });
+        const promiseRemoveService: Result<any, Error> = await promiseAsyncRemoveService;
+        if (!promiseRemoveService.isOk()) {
+            return promiseRemoveService.error;
+        }
+
+        winston.debug("Successfully removed service ID %v", serviceId);
+
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async repartitionNetwork(
```

Comment:
The return type of this function should be `Promise<Result<void>>`, letting Result handle the Error for us